### PR TITLE
Adds App Store Connect post

### PR DESCRIPTION
**Pull request by fgauchet**

### CODE REVIEW

File: _posts/2019-12-02-testflight-app-store-connect-api.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+layout: post
+title:  "Increasing TestFlight Adoption With the App Store Connect API"
+authors:
+  - "Frédéric Gauchet"
+excerpt: >
+  How Deliveroo uses the App Store Connect API to check if a new TestFlight build is available and prompt employees to install it.
+
+---
+
+At Deliveroo, we rely on TestFlight to ensure that the app ships to the App Store with as few issues as possible. We don’t have manual testers and since we operate in 13 countries and support a number of country specific features and payment options, it is important to get as many people as possible to install the latest beta to cover most use cases. To do this we prompt employees in app with a modal screen to invite them to install the latest beta.
```
Comment (K4KYA) [inline _posts/2019-12-02-testflight-app-store-connect-api.md]: Edit for clarity:
"ensure that the app" -> "ensure our iOS app"?

File: _posts/2019-12-02-testflight-app-store-connect-api.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+One use case that was particularly interesting to us was fetching the build number of the latest TestFlight public beta from App Store Connect. We came up with the following query to fetch the info we needed:
+
+```
+$ curl -g “https://api.appstoreconnect.apple.com/v1/builds?limit=10&sort=-version&filter[app]=<apple_app_id>&include=buildBetaDetail” --Header “Authorization: Bearer <generated JWT token>”
+```
+
+In the query above, replace `<apple_app_id>` with your own app ID.
+The JWT token can be generated with the script below, taken from the [WWDC video presentation](https://developer.apple.com/videos/play/wwdc2018-303/?time=2019) about the App Store Connect API.
+
+The curl request above returns some JSON that contains amongst other things, the build number of the latest ten betas. It is then simple enough to look for the highest version number that has the following state: `“externalBuildState”: “IN_BETA_TESTING”`. You’ll have to do a bit of processing: The response contains a “data” object which contains the version number and an “included” object which contains the build state. You can map the version number to a build state by joining on the opaque IDs.
```
Comment (ceeK) [inline _posts/2019-12-02-testflight-app-store-connect-api.md]: >  The response contains a “data” object which contains the version number and an “included” object which contains the build state.

This sounds like it uses JSONAPI? Might be worth mentioning that if that's the case.

Shame our Mapperoo library isn't open sourced as we built that just to deal with JSONAPI, could have been a good plug 😎

Comment (K4KYA) [inline _posts/2019-12-02-testflight-app-store-connect-api.md]: formatting:
"data" -> `data`
"included" -> `included`

File: _posts/2019-12-02-testflight-app-store-connect-api.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Authenticating with App Store Connect API
+
+Authentication is done with JSON Web Token which can be created with a private key created on App Store Connect, in the Users and Keys section. The key only needs Developer access, not Admin.
+
+As you must not embed a private key within an app we built a service to query the API. At the last [company hack day](/2019/10/02/where-are-they-now.html) we set out to build an AWS lambda (to avoid adding more code to our monolith backend) to fetch the version information.
```
Comment (K4KYA) [inline _posts/2019-12-02-testflight-app-store-connect-api.md]: Suggestion:
Remove "monolith"? It makes it sound like we only have the monolith! 😂

**Comment by fgauchet** [comment]
@K4KYA thanks for reviewing, I've addressed your comments now. Let me know if you have more suggestions :)

**Pull request merged**